lib/outline.js

```diff
@@ -4,14 +4,22 @@ function getOutline (node, depth) {
   if (depth === 0) {
     return []
   }
-
   return node.getSections()
     .filter(section => section.getId())
-    .map(section => ({
-      title: section.getTitle(),
-      destination: section.getId(),
-      children: getOutline(section, depth - 1)
-    }))
+    .map(section => {
+      const title = section.getTitle()
```

Comment:
the list of replacements is not complete

Reply:
Would you consider using a [package](https://www.npmjs.com/package/html-entities)?

Reply:
BTW, thanks so much for looking into this.

Reply:
I don't mind using the popular and dependency-free `html-entities` package but we will also need to encode the string to conform with the specification:

> Certain strings contain information that is intended to be human-readable, such
as text annotations, bookmark names, article names, document information, and
so forth. Such strings are referred to as text strings. Text strings are encoded in
either PDFDocEncoding or Unicode character encoding. PDFDocEncoding is a
superset of the ISO Latin 1 encoding and is documented in Appendix D. Unicode
is described in the Unicode Standard by the Unicode Consortium (see the Bibli-
ography).
For text strings encoded in Unicode, the first two bytes must be 254 followed by
255, representing the Unicode byte order marker, U+FEFF . (This sequence con-
flicts with the PDFDocEncoding character sequence thorn ydieresis , which is un-
likely to be a meaningful beginning of a word or phrase.) The remainder of the
string consists of Unicode character codes, according to the UTF-16 encoding
specified in the Unicode standard, version 2.0. Commonly used Unicode values
are represented as 2 bytes per character, with the high-order byte appearing first
in the string.

https://www.adobe.com/content/dam/acom/en/devnet/pdf/pdfs/pdf_reference_archives/PDFReference.pdf

Reply:
I thought the text wasn't properly encoded but it appears that the right double quotation mark is not rendered properly (at least on my machine): https://www.fileformat.info/info/unicode/char/201d/index.htm

![typo-quote-ide](https://user-images.githubusercontent.com/333276/69899742-08e6e700-136b-11ea-90d3-2d33616fd9e2.png)

I've tried to open the PDF in both [Evince](https://wiki.gnome.org/Apps/Evince) (the default Gnome PDF viewer) and Okular:


Evince | Okular
------------ | -------------
![typo-quote](https://user-images.githubusercontent.com/333276/69899743-08e6e700-136b-11ea-8f49-32f6542b3cd0.png) | ![okular](https://user-images.githubusercontent.com/333276/69899815-f5884b80-136b-11ea-9dd8-62314a23ffb6.png)

Reply:
It looks fine when using Asciidoctor PDF (Ruby):

![asciidoctor-pdf-ruby](https://user-images.githubusercontent.com/333276/69899912-0d140400-136d-11ea-8559-0930cb69d60c.png)

Reply:
I think I got it!
I'm now using `PDFHexString` to encode the string and I'm using a function to sanitize the string similar to what Asciidoctor PDF (Ruby) is doing:


```ruby
module Asciidoctor
module PDF
module Sanitizer
  (BuiltInNamedEntities = {
    'amp' => ?&,
    'apos' => ?',
    'gt' => ?>,
    'lt' => ?<,
    'nbsp' => ' ',
    'quot' => ?",
  }).default = ??
  SanitizeXMLRx = /<[^>]+>/
  CharRefRx = /&(?:([a-z][a-z]+\d{0,2})|#(?:(\d\d\d{0,4})|x([a-f\d][a-f\d][a-f\d]{0,3})));/

  # Strip leading, trailing and repeating whitespace, remove XML tags and
  # resolve all entities in the specified string.
  def sanitize string
    string = string.gsub(SanitizeXMLRx, '') if string.include? '<'
    string = string.gsub(CharRefRx) { $1 ? BuiltInNamedEntities[$1] : ([$2 ? $2.to_i : ($3.to_i 16)].pack 'U1') } if string.include? '&'
    string.strip.tr_s ' ', ' '
  end
end
```